provider/migration/base_steps.py

```
@@ -484,6 +487,41 @@ def remote_add_or_remove_port(self, port, add=True):
+        backup_uri, self.vm.connect_uri = self.vm.connect_uri, dest_uri
+        self.vm.cleanup_serial_console()
+        self.vm.create_serial_console()
+        vm_session = self.vm.wait_for_serial_login(timeout=120)
+        o = vm_session.cmd_output(f"md5sum {file_md5sum_path}").strip()
+        self.test.log.debug(f"File md5sum: {o}")
+        vm_session.close()
+        self.vm.connect_uri = backup_uri
+        if o != self.file_md5sum_in_vm:
+            self.test.fail(f"Expected md5sum {self.file_md5sum_in_vm}, but got {o}")
+        else:
+            self.test.log.info(f"Data integrity check passed.")
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)